src/__experimental__/components/input/input-presentation.spec.js

```diff
@@ -16,6 +25,100 @@ describe('InputPresentation', () => {
       .hasClass('carbon-input-presentation--has-focus')).toBeTruthy();
   });
 
+  describe('style', () => {
+    const {
+      components: { inputPresentation: inputStyleRules },
+      colors: {
+        error, warning, text
+      }
+    } = baseTheme;
+
+    describe('sizes', () => {
+      const { small, medium, large } = stripKeys(inputStyleRules.sizes, ['width', 'fontSize']);
+
+      it('has the right style for small-sized inputs', () => {
+        assertStyleMatch(small, mountRender({ size: 'small' }));
+      });
+      it('has the right style for medium-sized inputs', () => {
+        assertStyleMatch(medium, mountRender({ size: 'medium' }));
+      });
+      it('has the right style for large-sized inputs', () => {
+        assertStyleMatch(large, mountRender({ size: 'large' }));
+      });
+    });
+
+    describe('states', () => {
+      describe('disabled', () => {
+        it('has the correct style rules', () => {
+          assertStyleMatch(
+            {
+              color: text.disabled,
+              background: inputStyleRules.states.disabled.background,
+              cursor: 'not-allowed'
+            },
+            mountRender({ disabled: true })
+          );
+        });
+      });
+
+      describe('active', () => {
+        it('has the correct style rules', () => {
+          const wrapper = mountRender({ hasFocus: true });
+
+          const styleRules = {
+            color: text.body,
+            background: inputStyleRules.base.background,
+            outline: `3px solid ${warning}`,
+            border: `${inputStyleRules.base.border}`
+          };
+
+          assertStyleMatch(
+            styleRules,
+            wrapper
+          );
+        });
+      });
+
+      describe('readOnly', () => {
+        it('has the correct style rules', () => {
+          assertStyleMatch(
+            {
+              background: inputStyleRules.states.readOnly.background,
+              border: 'none'
+            },
+            mountRender({ readOnly: true })
+          );
+        });
+      });
+
+      describe('error', () => {
+        it('has the correct style rules', () => {
+          assertStyleMatch(
+            {
+              color: text.body,
+              background: inputStyleRules.base.background,
+              border: `2px solid ${error}`
+            },
+            mountRender({ error: true })
+          );
+        });
+      });
+
+      describe('warning', () => {
+        it('has the correct style rules', () => {
+          assertStyleMatch(
+            {
+              color: text.body,
+              background: inputStyleRules.base.background,
+              border: `2px solid ${warning}`
+            },
+            mountRender({ warning: true })
+          );
+        });
+      });
+    });
+  });
+
```

Comment:
do we really need to test all those styles? I think that snapshots should do the same thing.